contracts/MarketOracle.sol

```diff
@@ -58,10 +67,34 @@ contract MedianOracle is Ownable, IOracle {
 
     function pushReport(uint256 payload) external
     {
-        address sender = msg.sender;
-        require(providerReports[sender].timestamp > 0);
-        providerReports[sender].timestamp = now;
-        providerReports[sender].payload = payload;
+        address providerAddress = msg.sender;
+        Report[2] memory reports = providerReports[providerAddress];
```

Comment:
There could be a slight gas improvement by only reading the timestamps into memory.

Something like
uint256[2] memory timestamps = {providerReports[providerAddress][0].timestamp, providerReports[providerAddress][1].timestamp}

Reply:
Good one, will test it.

Reply:
This can make a 2% improvement 48866 vs 49759

```suggestion
    function pushReport(uint256 payload) external
    {
        address providerAddress = msg.sender;
        Report[2] storage reports = providerReports[providerAddress];
        uint256[2] memory timestamps = [reports[0].timestamp, reports[1].timestamp];

        require(timestamps[0] > 0);

        uint8 index_recent = timestamps[0] >= timestamps[1] ? 0 : 1;
        uint8 index_past = 1 - index_recent;

        uint256 nowTimestamp = now;

        // Check the push is not too soon after the last one.
        require(timestamps[index_recent].add(reportSecurityDelaySec) <= nowTimestamp);

        providerReports[providerAddress][index_past].timestamp = nowTimestamp;
        providerReports[providerAddress][index_past].payload = payload;
    }
```

Reply:
Which one do you like more?

Reply:
I think the second one (non-commented) is still pretty clean. Especially if we can remove the `nowTimestamp` var.

Reply:
We can it will add SLOAD operation. as now is surprisingly in storage not memory.